components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java

```diff
@@ -775,4 +778,17 @@ private AuthenticatorConfig getAuthenticatorConfig(String name) {
         }
         return authConfig;
     }
+
+    private JsGraphBuilderFactory createJsGraphBuilderFactory() {
+        String scriptEngineName = IdentityUtil.getProperty("AdaptiveAuth.ScriptEngine");
+        if (scriptEngineName != null) {
+            if ("graaljs".equals(scriptEngineName)) {
+                return new JsPolyglotGraphBuilderFactory();
+            }
+            return new JsNashornGraphBuilderFactory();
+        }
+        return null;
```

Comment:
If the config is not available, better to send the Nashorn factory, for backward compatibility.

Reply:
Modified